### PR TITLE
ie10-11 js fix

### LINKED ISSUE #2694: IE10 js error
opengrok 1.2.2
1. Object doesn't support property or method 'includes' -  utils-0.0.29.js:1787
2. Object doesn't support property or method "startsWith" -  utils-0.0.29.js:1868
3. Object doesn't support property or method "isInteger" - utils-0.0.29.js:1772

### PR DESCRIPTION
**Pull request by Teriand**
fixes #2694

### CODE REVIEW

**Comment by vladak** [comment]
Pls do the move of the file to the new version.

File: opengrok-web/src/main/webapp/js/utils-0.0.29.js [hunk 1]
```diff
@@ -2390,3 +2390,23 @@ function getCookie(cname) {
...
+    {
+        return typeof value === 'number' && isFinite(value) && Math.floor(value) === value;
+    };
+}
+
+if (!String.prototype.startsWith) {
+  String.prototype.startsWith = function(searchString, position) {
+    position = position || 0;
+    return this.indexOf(searchString, position) === position;
+  };
```
Comment (tulinkry) [inline opengrok-web/src/main/webapp/js/utils-0.0.29.js]: there's only one match, isn't it better to replace it inline?

File: opengrok-web/src/main/webapp/js/utils-0.0.29.js [hunk 2]
```diff
@@ -2390,3 +2390,23 @@ function getCookie(cname) {
...
+//polyfills for IE10-11
+// References:
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger
+
+
+if (!Number.isInteger)
+{
+    Number.isInteger = function numberIsInteger(value)
+    {
+        return typeof value === 'number' && isFinite(value) && Math.floor(value) === value;
```
Comment (tulinkry) [inline opengrok-web/src/main/webapp/js/utils-0.0.29.js]: there is only one match, isn't it better to do it inline or use some kind of alternative with parseInt() === NaN ?

**Comment by vladak** [comment]
Also, use `fixes #2694` in the commit message for autoclose on merge.

File: opengrok-web/src/main/webapp/js/utils-0.0.29.js [hunk 3]
```diff
@@ -1754,7 +1754,7 @@ function initAutocomplete(config, minisearch) {
 }
 
 function initMinisearchAutocomplete(config) {
-    if (config.allowedFields && !config.allowedFields.includes('full')) {
+    if (config.allowedFields && config.allowedFields.indexOf('full')<0) {
```
Comment (tulinkry) [inline opengrok-web/src/main/webapp/js/utils-0.0.29.js]: there should be a space `ull')<0)` -> `ull') < 0)`

File: opengrok-web/src/main/webapp/js/utils-0.0.29.js [hunk 4]
```diff
@@ -1784,7 +1784,7 @@ function initMinisearchAutocomplete(config) {
 }
 
 function initAutocompleteForField(inputId, field, config, dataFunction, errorElemId) {
-    if (config.allowedFields && !config.allowedFields.includes(field)) {
+    if (config.allowedFields && config.allowedFields.indexOf(field)<0) {
```
Comment (tulinkry) [inline opengrok-web/src/main/webapp/js/utils-0.0.29.js]: there should be a space `field')<0)` -> `field') < 0)`

Comment (Teriand) [reply opengrok-web/src/main/webapp/js/utils-0.0.29.js]: thx, fixed

File: opengrok-web/src/main/webapp/js/utils-0.0.29.js (hunk 3 above)
Comment (Teriand) [reply opengrok-web/src/main/webapp/js/utils-0.0.29.js]: thx, fixed

File: opengrok-web/src/main/webapp/js/utils-0.0.29.js (hunk 1 above)
Comment (Teriand) [reply opengrok-web/src/main/webapp/js/utils-0.0.29.js]: i can change to simple
```
if (event.originalEvent.originalEvent.type.indexOf('key') == 0) {
```

File: opengrok-web/src/main/webapp/js/utils-0.0.29.js (hunk 2 above)
Comment (Teriand) [reply opengrok-web/src/main/webapp/js/utils-0.0.29.js]: if caretPos need only simple number check,
i can change to 
```
typeof caretPos === 'number'
```

**Comment by tulinkry** [comment]
Looks ok, can we merge?

**Pull request merged**